components/postcode-lookup/index.js

```diff
@@ -0,0 +1,339 @@
+
+          // Different addresses have different properties returned by the API
+          let addressLine1 = '';
+          let addressLine2 = '';
+          if (obj.ORGANISATION_NAME !== undefined && obj.BUILDING_NAME !== undefined &&
+            obj.BUILDING_NUMBER !== undefined) {
+            addressLine1 = obj.ORGANISATION_NAME + ', ' + obj.BUILDING_NAME;
+            addressLine2 = obj.BUILDING_NUMBER + ' ' + obj.THOROUGHFARE_NAME;
+          } else if (obj.ORGANISATION_NAME !== undefined && obj.BUILDING_NAME !== undefined) {
+            addressLine1 = obj.ORGANISATION_NAME + ', ' + obj.BUILDING_NAME;
+            addressLine2 = obj.THOROUGHFARE_NAME;
+          } else if (obj.ORGANISATION_NAME !== undefined && obj.BUILDING_NUMBER !== undefined) {
+            addressLine1 = obj.ORGANISATION_NAME;
+            addressLine2 = obj.BUILDING_NUMBER + ' ' + obj.THOROUGHFARE_NAME;
+          } else if (obj.BUILDING_NAME !== undefined && obj.THOROUGHFARE_NAME !== undefined) {
+            addressLine1 = obj.BUILDING_NAME + ' ' + obj.THOROUGHFARE_NAME;
+          } else {
+            addressLine1 = obj.BUILDING_NUMBER + ' ' + obj.THOROUGHFARE_NAME;
+          }
+
+          const townOrCity = obj.POST_TOWN;
+          // Get element ID to be used in determining if radio button is checked in the saveValues section
+          const radioButtonId = `${addressKey}-select-${obj.ADDRESS}`;
+          return {
+            elementId: radioButtonId,
+            label: addressValue,
+            value: addressValue,
+            addressLine1: addressLine1,
+            addressLine2: addressLine2,
+            townOrCity: townOrCity
+          };
+        }));
+
+        req.form.options.fields[`${addressKey}-select`] = {
+          mixin: 'radio-group',
+          legend: {
+            className: 'visuallyhidden'
+          },
+          className: 'govuk-radios--inline',
+          validate: ['required'],
+          options: req.sessionModel.get('addressesOptions')
+        };
+        req.sessionModel.set('addressCount', addresses.length);
+      }
+      super.getValues(req, res, callback);
+    }
+
+    locals(req, res, callback) {
+      const isManual = req.query.step === 'manual';
+      const locals = super.locals(req, res, callback);
+      const sessionPostcode = req.sessionModel.get(`${addressKey}-postcode`);
+      const section = this.options.route.replace(/^\//, '');
+      const editLink = conditionalTranslate('pages.address-lookup.edit', req.translate) || defaults.CHANGE;
+      const searchByPostcodeLink = conditionalTranslate('pages.address.searchByPostcode', req.translate) ||
+      defaults.SEARCH_BY_POSTCODE;
+      const cantFind = conditionalTranslate('pages.address-lookup.cantfind', req.translate) || defaults.CANT_FIND;
+
+      let postcodeApiMessageKey;
+      let postcodeError;
+
+      if (!isManual) {
+        postcodeApiMessageKey = (req.sessionModel.get(`${addressKey}-postcodeApiMeta`) || {}).messageKey;
```

Comment:
Removed the postcodeApiMeta value altogether now since we are not displaying an error on the screen, but instead redirecting to an error screen using the response codes returned by the API